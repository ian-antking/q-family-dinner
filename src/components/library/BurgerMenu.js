import React, { useState, useRef } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import useOnClickOutside from '../../hooks/hooks';
import Burger from './BurgerMenuComponents/Burger';
import Menu from './BurgerMenuComponents/Menu';

const StyledBurgerMenu = styled.div`
  height: 50px;
  width: 60px;
`;

const BurgerMenu = ({ content }) => {
  const [open, setOpen] = useState(false);
  const node = useRef();
  useOnClickOutside(node, () => setOpen(false));

  return (
    <StyledBurgerMenu data-testid="burgermenu" ref={node}>
      <Burger open={open} setOpen={setOpen} />
      <Menu open={open} setOpen={setOpen} content={content} />
    </StyledBurgerMenu>
  );
};

BurgerMenu.propTypes = {
  content: PropTypes.shape.isRequired,
};

export default BurgerMenu;
