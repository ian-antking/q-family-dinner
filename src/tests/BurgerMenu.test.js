import React from 'react';
import { HashRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import BurgerMenu from '../components/library/BurgerMenu';

const content = {
  pages: [
    {
      fields: {
        title: "test-title",
        slug: "test-slug",
        primaryPage: true
      },
      sys: {
        id: "test-id"
      }
    }
  ]
};

test('snapshot: BurgerMenu component', () => {
  const { asFragment } = render(
    <Router>
      <BurgerMenu content={content}/>
    </Router>
  );
  const component = asFragment();
  expect(component).toMatchSnapshot();
});

test('BurgerMenu component renders', () => {
  const { getByTestId } = render(
    <Router>
      <BurgerMenu content={content}/>
    </Router>
  );
  const componentId = getByTestId('burgermenu');
  expect(componentId).toBeInTheDocument();
});