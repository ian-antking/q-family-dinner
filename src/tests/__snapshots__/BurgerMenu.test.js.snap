// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot: BurgerMenu component 1`] = `
<DocumentFragment>
  <div
    class="sc-hKFyIo lnHSxj"
    data-testid="burgermenu"
  >
    <button
      class="sc-bdnylx bimBLs"
      data-testid="burger"
    >
      <div
        data-testid="slice-one"
      />
      <div
        data-testid="slice-two"
      />
      <div
        data-testid="slice-three"
      />
    </button>
    <nav
      class="sc-gtssRu gtXGLq"
      data-testid="menu"
    >
      <a
        class="sc-dlnjPT gpLYWd"
        href="#/test-slug"
      >
        test-title
      </a>
    </nav>
  </div>
</DocumentFragment>
`;
